### PR TITLE
[DAT-17482] Integrating Bigquery into liquibase

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddColumnExecutorTest.java [hunk 1]
```diff
@@ -53,6 +54,7 @@ public void generateSql_notNull() throws Exception {
         assertCorrect("alter table [table_name] add [column_name] [int] constraint df_table_name_column_name default 42 not null", MSSQLDatabase.class);
         assertCorrect("alter table table_name add column_name int default 42 not null", MySQLDatabase.class);
         assertCorrect("not supported. fixme!!", SQLiteDatabase.class);
+        assertCorrect("alter table [table_nam]e add column [column_name] int default 42 not null", BigqueryDatabase.class);
```
Comment (KushnirykOleh) [inline liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddColumnExecutorTest.java:57]: seems to be a typo

File: liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java [hunk 2]
```diff
@@ -158,7 +159,7 @@ public void execute_withSchema() throws Exception {
 
         assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", HsqlDatabase.class);
         assertCorrect("alter table \"lbcat2\".[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", DB2Database.class, Db2zDatabase.class);
-        assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", H2Database.class);
+        assertCorrect("alter table [lbschem2].[adduqtest] add constraint [uq_test] unique ([coltomakeuq])", H2Database.class, BigqueryDatabase.class);
```
Comment (KushnirykOleh) [inline liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java:162]: unique constraints are not supported by bigQuery

Comment (filipelautert) [reply liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java:162]: That's true! hm.. but when using the uniqueCosntraint tag the sql is being generated by core. So maybe we should we add a BigqueryUniqueConstraintGenerator that throws an exception?

Comment (filipelautert) [reply liquibase-integration-tests/src/test/java/liquibase/statementexecute/AddUniqueConstraintExecutorTest.java:165]: Great, the problem is the test. I added a comment explaining it.

File: liquibase-bigquery/src/main/java/liquibase/sqlgenerator/BigqueryCreateDatabaseChangeLogLockTableGenerator.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import liquibase.database.Database;
+import liquibase.database.ObjectQuotingStrategy;
+import liquibase.datatype.DataTypeFactory;
+import liquibase.sql.Sql;
+import liquibase.sqlgenerator.core.CreateDatabaseChangeLogLockTableGenerator;
+import liquibase.statement.NotNullConstraint;
+import liquibase.statement.core.CreateDatabaseChangeLogLockTableStatement;
+import liquibase.statement.core.CreateTableStatement;
+
+public class BigqueryCreateDatabaseChangeLogLockTableGenerator extends CreateDatabaseChangeLogLockTableGenerator {
```
[the remark this answers is not in the document]
Comment (filipelautert) [reply liquibase-bigquery/src/main/java/liquibase/sqlgenerator/BigqueryCreateDatabaseChangeLogLockTableGenerator.java]: Done!

File: liquibase-bigquery/src/main/java/liquibase/snapshot/jvm/BigQuerySequenceSnapshotGenerator.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                return PRIORITY_NONE;
+            }
+        }
+
+        @Override
+        protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException {
+            Scope.getCurrentScope().getLog(this.getClass()).info("Sequences are not supported by BigQuery");
+            return null;
+        }
+
```
[the remark this answers is not in the document]
Comment (filipelautert) [reply liquibase-bigquery/src/main/java/liquibase/snapshot/jvm/BigQuerySequenceSnapshotGenerator.java:26]: Makes sense... seems it is conflicting with the new implementation.

**Pull request closed without merging**